framework/rbac/DbManager.php

```diff
@@ -600,6 +608,10 @@ public function getAssignment($roleName, $userId)
         if (empty($userId)) {
             return null;
         }
+        if ($this->cacheAssignments) {
```

Comment:
It will populate the cache and return a single result from it. If the number of assignments for a single user is not big there won't be any significant difference in the query speed as the execution itself has larger overhead than fetching data. Obviously I'd need a benchmark to back it up but I've been making such tests in Yii 1.